src/TableExtension.java

```diff
@@ -343,6 +383,43 @@ public Object report(Argument args[], Context context)
 
   }
 
+  public static class FromJSON implements Reporter {
+    public Syntax getSyntax() {
+      return SyntaxJ.reporterSyntax
+          (new int[]{Syntax.StringType()},
+              Syntax.WildcardType());
+    }
+
+    public String getAgentClassString() {
+      return "OTPL";
+    }
+
+    public Object report(Argument args[], Context context)
+        throws ExtensionException, LogoException {
+
+      FileManager fm = ((ExtensionContext) context).workspace().fileManager();
+
+      try {
+        String path = fm.attachPrefix(args[0].getString());
+        File file = new File(path.toString());
+        if (!file.exists()) {
+          throw new ExtensionException (args[0].get().toString() + " does not exist.");
+        }
+        Gson gson = new Gson();
+
+        try {
+          java.util.Map<Object,Object> map = gson.fromJson(new FileReader(path), java.util.Map.class);
+          return new Table(map);
+        } catch (com.google.gson.JsonSyntaxException e) {
+          throw new ExtensionException ("Error trying to read the JSON file. It is probably missing a colon or comma. See the line number on the next line: " + e.getMessage());
```

Comment:
Prefer no space between a constructor and its open paren, so `ExtensionException(`